src/object_store/src/object/opendal_engine/opendal_s3.rs

```diff
@@ -65,6 +65,54 @@ impl OpendalObjectStore {
         })
     }
 
+    /// Creates a minio client. The server should be like `minio://key:secret@address:port/bucket`.
+    pub fn with_minio(server: &str, object_store_config: ObjectStoreConfig) -> ObjectResult<Self> {
+        let server = server.strip_prefix("minio://").unwrap();
+        let (access_key_id, rest) = server.split_once(':').unwrap();
+        let (secret_access_key, mut rest) = rest.split_once('@').unwrap();
+        let endpoint_prefix = if let Some(rest_stripped) = rest.strip_prefix("https://") {
+            rest = rest_stripped;
+            "https://"
+        } else if let Some(rest_stripped) = rest.strip_prefix("http://") {
+            rest = rest_stripped;
+            "http://"
+        } else {
+            "http://"
+        };
+        let (address, bucket) = rest.split_once('/').unwrap();
```

Comment:
I guess it can be better to use the [`url`](https://docs.rs/url/latest/url/) crate to parse the URL here instead of doing it manually. Less code and less possibility of security problems.

Reply:
Umm just tried, but the here it's not a normal url(`127.0.0.1:9301/hummock001`, instead of `http://127.0.0.1:9301/hummock001`), so just keep it the same as [before](https://github.com/risingwavelabs/risingwave/blob/6a401d7fe4daded3478439523a68877c4316f982/src/object_store/src/object/s3.rs#L661).

Reply:
"minio://" is part of the URL

Reply:
Let's change the use of all state store urls in later PR.